packages/popup/src/pages/ConnectSite/index.js

```diff
@@ -0,0 +1,243 @@
+            <div key={groupIndex}>
+              <p className="text-gray-40 ml-4 mb-1 mt-1 text-xs">{nickname}</p>
+              {account.map(({nickname, eid, address}, accountIndex) => (
+                <div
+                  aria-hidden="true"
+                  onClick={() => () => {}}
+                  key={accountIndex}
+                  className="flex px-3 items-center h-15"
+                >
+                  <div className="flex w-full">
+                    <Avatar
+                      className="w-5 h-5 mr-2"
+                      diameter={20}
+                      accountId={eid}
+                    />
+                    <div className="flex-1">
+                      <p className="text-xs text-gray-40">{nickname}</p>
+                      <p className="text-sm text-gray-80">
+                        {address ? shortenAddress(address) : ''}
+                      </p>
+                    </div>
+                    <div className="flex">
+                      {currentAccount?.eid === eid ? (
+                        <img
+                          src="images/location.svg"
+                          alt="current address"
+                          className="mr-3"
+                        />
+                      ) : null}
+                      <Checkbox
+                        onChange={() => onSelectSingleAccount(eid)}
+                        checked={checkboxStatusObj[eid]}
+                      />
+                    </div>
+                  </div>
+                </div>
+              ))}
+            </div>
+          ))}
+        </div>
+      </CompWithLabel>
+    </>
+  ) : null
+}
+ConnectSitesList.propTypes = {
+  accountData: PropTypes.array.isRequired,
+  allCheckboxStatus: PropTypes.bool.isRequired,
+  checkboxStatusObj: PropTypes.object.isRequired,
+  currentAccount: PropTypes.object.isRequired,
+  onSelectAllAccount: PropTypes.func.isRequired,
+  onSelectSingleAccount: PropTypes.func.isRequired,
+}
+
+function ConnectSite() {
+  const {t} = useTranslation()
+  const [confirmAccounts, setConfirmAccounts] = useState([])
+  const [searchContent, setSearchContent] = useState('')
+  const [networkShow, setNetworkShow] = useState(false)
+  const [networkId, setNetworkId] = useState(null)
+  const [searchIcon, setSearchIcon] = useState('')
+  const [checkboxStatusObj, setCheckboxStatusObj] = useState({})
+  const [allCheckboxStatus, setAllCheckboxStatus] = useState(false)
+
+  const {addressDataWithAccountId, accountData} =
+    useAccountGroupAddress(networkId)
+  const {data: currentNetworkData} = useRPC([GET_CURRENT_NETWORK])
+  const {data: currentAccount} = useRPC([GET_CURRENT_ACCOUNT], undefined, {
+    fallbackData: {},
+  })
+  useDeepCompareEffect(() => {
+    setSearchIcon(currentNetworkData?.icon || '')
+    setSearchContent(currentNetworkData?.name || '')
+    setNetworkId(currentNetworkData?.eid || null)
+  }, [currentNetworkData || {}])
```

Comment:
don't use useDeepCompareEffect
this page only rerenders on
1. user change network
2. user click on check box
3. user click on cancel or connect
all of which are visible state in this component
there's no need to use useDeepCompareEffect